### ISSUE TITLE
Change default of TRANSFORM_INVALID_GROUP_CHARS from 'never' to 'ignore'

### ISSUE THREAD

**Issue by geerlingguy** [ansible/ansible#70908]
##### SUMMARY

After the long and drawn out issue https://github.com/ansible/ansible-documentation/issues/89 was mostly concluded by un-deprecating dashes in group names, I noticed that my Ansible installations are still spitting out warnings by default when using inventory plugins that build group names based on things like AWS regions.

```
[WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details
```

I propose that the default for `TRANSFORM_INVALID_GROUP_CHARS` be changed from `'never'` to `'ignore'`.

It's also unclear to me what value there is in having a `'never'` option, since it just does nothing of substance... and then spits out a warning? It seems like a scary message indicating something could break if I ignore it, but nothing _does_ break, so I wonder what's the point of displaying the message?

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

lib/ansible/inventory/group.py

##### ANSIBLE VERSION
```paste below
ansible 2.9.10
  config file = /Users/jgeerling/Development/GitHub/ansible-for-devops/lamp-infrastructure/ansible.cfg
  configured module search path = ['/Users/jgeerling/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.7 (default, Apr  9 2020, 17:13:06) [Clang 11.0.3 (clang-1103.0.32.29)]
```

##### CONFIGURATION

```paste below
ANSIBLE_NOCOWS(/Users/jgeerling/Development/GitHub/ansible-for-devops/lamp-infrastructure/ansible.cfg) = True
DEFAULT_ROLES_PATH(/Users/jgeerling/Development/GitHub/ansible-for-devops/lamp-infrastructure/ansible.cfg) = ['/Users/jg
HOST_KEY_CHECKING(/Users/jgeerling/Development/GitHub/ansible-for-devops/lamp-infrastructure/ansible.cfg) = False
RETRY_FILES_ENABLED(/Users/jgeerling/Development/GitHub/ansible-for-devops/lamp-infrastructure/ansible.cfg) = False
```

##### OS / ENVIRONMENT
macOS 10.15.5.

##### STEPS TO REPRODUCE

`inventory.ini` contents:

```
[my-hosts]
127.0.0.1 ansible_connection=local
```

`main.yml` contents:

```yaml
- hosts: my-hosts
  tasks:
    - meta: noop
```

Run the playbook with: 

##### EXPECTED RESULTS

No warnings appear.

##### ACTUAL RESULTS

```paste below
$ ansible-playbook -i inventory.ini main.yml 
[WARNING]: Invalid characters were found in group names but not replaced, use -vvvv to see details

PLAY [my-hosts] *******************************************
...
```

When I run with `-vvvv` it says:

```
Not replacing invalid character(s) "{'-'}" in group name (my-hosts)
```

But I'm not sure how that's actionable or why it matters enough to throw a warning.

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/plugins/inventory/openstack.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/inventory/openstack.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by geerlingguy** [comment]
!component = lib/ansible/inventory/group.py

**Comment by ansibot** [comment]
Files identified in the description:
* [``](https://github.com/ansible/ansible/blob/devel/)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [``](https://github.com/ansible/ansible/blob/devel/)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [``](https://github.com/ansible/ansible/blob/devel/)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [``](https://github.com/ansible/ansible/blob/devel/)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jamescassell** [comment]
I support this change.

**Comment by ansibot** [comment]
Files identified in the description:
* [``](https://github.com/ansible/ansible/blob/devel/)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/inventory/group.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/inventory/group.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sivel** [comment]
This horse has been beat to something that no longer looks like a horse.  We've discussed this again in our triage meeting, and we believe that the current configuration addresses both sides of this concern. However neither side is necessarily 100% happy with the outcome.  I believe we have concluded the zero-sum game here.

We warn to let users know they may have issues with invalid chars, but we don't implicitly modify those entries any longer.
